fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/DimensionsKeyExpression.java

```diff
@@ -89,7 +89,13 @@ public boolean createsDuplicates() {
 
     @Override
     public List<Descriptors.FieldDescriptor> validate(@Nonnull Descriptors.Descriptor descriptor) {
-        return getWholeKey().validate(descriptor);
+        Preconditions.checkArgument(prefixSize + dimensionsSize <= wholeKey.getColumnSize());
```

Comment:
Should throw `InvalidExpressionException` and not `IllegalArgumentException`.

Reply:
Done.